### PR TITLE
Support yes and no for node object attribute consoleondemand

### PR DESCRIPTION
**Pull request by gurevichmark**
This pull request fixes the attribute value inconsistency of `consoleondemad` attribute in `nodehm` table. Reported by issue #1801 

1. Verifies that `consoleondemad` attribute in `site` table is set to "yes" or "no". Error is reported if something else.
2. Allows case insensitive setting for `consoleondemad` attribute in `site` table
3. Allows case insensitive setting for `consoleondemad` attribute in `nodehm` table to "yes", "no" in addition to existing 0, or 1

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/conserver.pm [hunk 1]
```diff
@@ -575,10 +577,11 @@ sub donodeent {
             }
         }
         if (defined($cfgent->{consoleondemand})) {
-            if ($cfgent->{consoleondemand} && !$siteondemand) {
+            # consoleondemand attribute for node can be "1", "yes", "0" and "no"
+            if ((($cfgent->{consoleondemand} eq "1") || lc($cfgent->{consoleondemand}) eq "yes") && !$siteondemand) {
                 push @$content, "  options ondemand;\n";
             }
-            elsif (!$cfgent->{consoleondemand} && $siteondemand) {
+            elsif ((($cfgent->{consoleondemand} eq "0") || lc($cfgent->{consoleondemand}) eq "no") && $siteondemand) {
```
Comment (daniceexi) [inline xCAT-server/lib/xcat/plugins/conserver.pm]: @gurevichmark 
I think the code logic here is a little strange. As mentioned in the definition of node.consoleondemand:
~~~~
consoleondemand:	This overrides the value from site.consoleondemand; (0=no, 1=yes). Default is the result from site.consoleondemand.
~~~~

The setting in the ``node.consoleondemand`` has higher priority. If it was set, we don't need to check the ``$siteondemand``.

And we need to change the man page for ``site.consoleondemand`` to support **yes|on**.

Comment (gurevichmark) [inline xCAT-server/lib/xcat/plugins/conserver.pm]: 1. Yes, I agree that those lines are confusing, but I think it is done this way so as to not repeat the settings from `default` section of `/etc/conserver.cf`
`/etc/conserver.cf` file has a section for `default` and a section for each node.
For example, if `site.consoleondemand` is set to `yes`, we will put a line `option ondemand` into the `default` section. Then if `nodehm.consoleondemand` is also set to `yes`, the way the code is written today, we will not insert the same line into the individual node section of `/etc/conserver.cf` file.
If I change the code to only check the setting of `nodehm.consoleondemand`, we could repeat the same line from `default` section into the individual node section.
2. I made a second commit to update the `Schema.pm` file which is used to generate man pages containing `consoleondemand` attribute.

Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/conserver.pm]: Shouldn't the node overwrite go into each node section of the `/etc/conserver.cf` file? maybe i'm not following this conv very well...

**Comment by daniceexi** [comment]
Then I am fine with the change and agree to merge.

**Pull request merged**